src/pytest_allure_spec_coverage/spec_collectors/sphinx.py

```diff
@@ -0,0 +1,117 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Sphinx spec collector implementation"""
+import os
+import warnings
+from typing import Optional
+from docutils.core import publish_doctree
+from docutils.utils import SystemMessage
+
+from pytest_allure_spec_coverage.models.collector import Collector
+from pytest_allure_spec_coverage.models.scenario import Scenario, Parent
+
+
+class SphinxCollector(Collector):
+    """
+    Sphinx collector allows you to collect scenarios from .rst files
+    At the moment, only title will be read from .rst file
+    Template variables for title do not support
+
+    Possible options:
+        - sphinx_dir: required option where is places your scenarios
+        - sphinx_endpoint: where is hosted your scenarios. Need to form scenario link. Optionally
+    """
+
+    sphinx_dir: str
+    spec_endpoint: Optional[str]
+
+    def validate_config(self):
+        if "sphinx_dir" not in self.config:
+            raise ValueError("Option sphinx_dir is required")
+        sphinx_dir = self.config.get("sphinx_dir")
+        if not os.path.exists(sphinx_dir):
+            raise ValueError(f"Directory with sphinx specs {sphinx_dir} doesn't exists")
+
+        self.sphinx_dir = sphinx_dir
+        self.spec_endpoint = self.config.get("spec_endpoint")
+
+    def collect(self):
+        branch = os.getenv("BRANCH")
+        scenarios = []
+        parents_display_names = {}
+        root_path = self.sphinx_dir.rsplit("/", maxsplit=1)[0]
+        for root, _, files in os.walk(self.sphinx_dir):
```

Comment:
How directories starting with `_` will be handled in this case?
They are not about scenarios, just helpers for configuration files, common parts, and stuff like that.
We may want an abstract function that will make a name filter for ignoring file and directory names.

Reply:
This kind of files can be placed in scenarios directory, not only outside of it

Reply:
do you have examples of such files in our specs?

Reply:
In ADB project